image-loader/app/lib/Downloader.scala

```diff
@@ -22,25 +25,34 @@ class Downloader(implicit ec: ExecutionContext) {
     val request = new Request.Builder().url(uri.toString).build()
     val response = client.newCall(request).execute()
 
-    val expectedSize = response.header("Content-Length").toInt
-    val input = response.body().byteStream()
+    val maybeExpectedSize = Try{response.header("Content-Length").toInt}.toOption
```

Comment:
Why do we `.toOption` this?

Reply:
`.toOption` means we can pattern match, which looks a bit nicer (preference). The alternative would be:

```scala
val maybeExpectedSize: Try = Try{response.header("Content-Length").toInt}

if (maybeExpectedSize.isFailure) {
  // log and throw
}
else {
  // happy path
}

```

Reply:
I thought it was possible to do something like:

```scala
val maybeExpectedSize: Try = Try{response.header("Content-Length").toInt}

maybeExpectedSize match {
 case Success(value) => doSomethingWith(value)
 case Failure(e) => ...
}
```

Reply:
🤦‍♂ yeah, that's better isn't it!